### PR TITLE
Add FormValidityProvider & tests 🚀.

### PR DESCRIPTION
**Pull request by yamafaktory**
In order to make testing fun & easier, I pushed Enzyme into the Jest pipeline.

Checklist:

- [x] Test added / Snapshots updated
- [x] Documentation added / updated

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -30,10 +32,13 @@
     "react/jsx-no-target-blank": 0,
     "react/no-array-index-key": 2,
     "react/no-multi-comp": [1, { "ignoreStateless": true }],
-    "react/self-closing-comp": [2, {
-      "component": true,
-      "html": true
-    }],
+    "react/self-closing-comp": [
```
Comment (yamafaktory) [inline .eslintrc]: Some noise coming from Prettier...

File: src/molecules/TextInput.jsx [hunk 2]
```diff
@@ -121,7 +121,16 @@ class TextInput extends React.Component {
...
+    const { STATES, validity } = this.context
+    const message = e.target.validationMessage
+    let hasState = false
+    for (const state of STATES) {
+      if (e.target.validity[state]) {
+        this.setState({ message: validity[state] || message })
+        break
+      }
+    }
+    hasState && this.setState({ message })
```
Comment (mosch) [inline src/molecules/TextInput.jsx]: I think this can `hasState` will never be true - do I miss something?

File: src/molecules/TextInput.jsx [hunk 3]
```diff
@@ -121,7 +121,16 @@ class TextInput extends React.Component {
 
   handleInvalid = e => {
     e.preventDefault()
-    this.setState({ message: e.target.validationMessage })
+    const { STATES, validity } = this.context
```
Comment (mosch) [inline src/molecules/TextInput.jsx]: Does this work if no context is provided at all?

File: src/molecules/TextInput.jsx (hunk 2 above)
Comment (yamafaktory) [reply src/molecules/TextInput.jsx]: Yeah, I forgot to add it... but it was in my mind :D

File: src/molecules/TextInput.jsx (hunk 3 above)
Comment (yamafaktory) [reply src/molecules/TextInput.jsx]: Should, let me test it!

Comment (yamafaktory) [reply src/molecules/TextInput.jsx]: Fixed in last commit, added a test for it.

File: src/behaviour/FormValidityProvider.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+/**
```
Comment (mosch) [inline src/behaviour/FormValidityProvider.jsx]: For proper doc generating (check out yarn build:docs) I guess the docs need to be in the line before the class declaration see 👇🏻

File: src/behaviour/FormValidityProvider.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  'tooShort',
+  'typeMismatch',
+  'valueMissing',
+]
+
+const validityPropTypes = STATES.reduce((acc, prop) => ({
+  ...acc,
+  [prop]: PropTypes.string,
+}), {})
+
```
Comment (mosch) [inline src/behaviour/FormValidityProvider.jsx]: 👈🏽 here

Comment (yamafaktory) [reply src/behaviour/FormValidityProvider.jsx]: Done.

**Pull request merged**